newIDE/app/src/MainFrame/EditorContainers/HomePage/BuildSection/utils.js

```diff
@@ -164,24 +164,88 @@ const formatExampleShortHeaderForCarousel = ({
   };
 };
 
+const formatItemForCarousel = ({
+  item,
+  onSelectGameTemplate,
+  onSelectExample,
+  i18n,
+  receivedGameTemplates,
+}: {
+  item: PrivateGameTemplateListingData | ExampleShortHeader,
+  onSelectGameTemplate: PrivateGameTemplateListingData => void,
+  onSelectExample: ExampleShortHeader => void,
+  i18n: I18nType,
+  receivedGameTemplates: ?Array<PrivateGameTemplate>,
+}): CarouselThumbnail => {
+  if (item.previewImageUrls) {
+    return formatExampleShortHeaderForCarousel({
+      exampleShortHeader: item,
+      onSelectExample: onSelectExample,
+    });
+  } else {
+    return formatGameTemplateListingDataForCarousel({
+      i18n,
+      onSelectGameTemplate: onSelectGameTemplate,
+      gameTemplateListingData: item,
+      receivedGameTemplates: receivedGameTemplates,
+    });
+  }
+};
+
+const formatItemForGrid = ({
+  item,
+  onSelectGameTemplate,
+  onSelectExample,
+  i18n,
+  receivedGameTemplates,
+}: {
+  item: PrivateGameTemplateListingData | ExampleShortHeader,
+  onSelectGameTemplate: PrivateGameTemplateListingData => void,
+  onSelectExample: ExampleShortHeader => void,
+  i18n: I18nType,
+  receivedGameTemplates: ?Array<PrivateGameTemplate>,
+}): React.Node => {
+  if (item.previewImageUrls) {
+    return (
+      <ExampleTile
+        exampleShortHeader={item}
+        onSelect={() => onSelectExample(item)}
+      />
+    );
+  } else {
+    const isTemplateOwned =
+      !!receivedGameTemplates &&
+      !!receivedGameTemplates.find(
+        receivedGameTemplate => receivedGameTemplate.id === item.id
+      );
+    return (
+      <PrivateGameTemplateTile
+        privateGameTemplateListingData={item}
+        onSelect={() => onSelectGameTemplate(item)}
+        owned={isTemplateOwned}
+      />
+    );
+  }
+};
+
 /**
  * This method allocates examples and private game templates between the
  * build section carousel and grid.
  * `numberOfItemsExclusivelyInCarousel` controls the number of items that
  * should appear in the carousel only. The rest appears in both the carousel
  * and the grid.
  */
-export const getExampleAndTemplateItemsForBuildSection = ({
+export const getExampleAndTemplateTiles = ({
```

Comment:
We have created a monster 😅

Reply:
haha, I've modified it and I feel it's simpler to understand now / and easier to modify. 🤞